### PR TITLE
issue #3509 - initialize cosClient before its first use

### PR DESCRIPTION
**Pull request by lmsurpre**
and log the full exception if there's an issue during COS scanning

Signed-off-by: Lee Surprenant <lmsurpre@us.ibm.com>

### CODE REVIEW

File: fhir-bucket/src/main/java/com/ibm/fhir/bucket/app/Main.java [hunk 1]
```diff
@@ -992,12 +992,14 @@ protected void scanAndLoad() {
 
             if (!this.cosBucketList.isEmpty()) {
                 // Process using COS as our source data repository
-                if (cosProperties != null && cosProperties.size() > 0) {
-                    cosClient = new COSClient(cosProperties);
-                } else {
-                    throw new IllegalArgumentException("COS configuration required");
+                if (cosClient == null) {
```
Comment (punktilious) [inline fhir-bucket/src/main/java/com/ibm/fhir/bucket/app/Main.java:995]: I wonder if we should have the COSClient initialization just in one place. The issue is how we decide whether or not we need it. Perhaps a method that creates it on demand if it doesn't exist?

File: fhir-bucket/src/main/java/com/ibm/fhir/bucket/scanner/CosScanner.java [hunk 2]
```diff
@@ -149,7 +153,7 @@ public void mainLoop() {
                 }
             } catch (Exception x) {
                 // Just catch and log so we don't break the main loop
-                logger.severe("Error during COS scan: " + x.getMessage());
+                logger.log(Level.SEVERE, "Error during COS scan", x);
```
Comment (punktilious) [inline fhir-bucket/src/main/java/com/ibm/fhir/bucket/scanner/CosScanner.java:156]: I think I was finding the stack traces annoying :-)

File: fhir-bucket/src/main/java/com/ibm/fhir/bucket/app/Main.java (hunk 1 above)
Comment (lmsurpre) [reply fhir-bucket/src/main/java/com/ibm/fhir/bucket/app/Main.java:995]: my we just add it to the current `configure()` method an instantiate it whenever we have some `cosProperties` passed to us?

in main, configure() is called right before process(), and that is where we set up the db stuff already...

File: fhir-bucket/src/main/java/com/ibm/fhir/bucket/scanner/CosScanner.java (hunk 2 above)
Comment (lmsurpre) [reply fhir-bucket/src/main/java/com/ibm/fhir/bucket/scanner/CosScanner.java:156]: hopefully we've fixed the cases where we see this over and over and want to ignore it... without the stack trace its really hard to know what went wrong

File: fhir-bucket/src/main/java/com/ibm/fhir/bucket/app/Main.java (hunk 1 above)
Comment (lmsurpre) [reply fhir-bucket/src/main/java/com/ibm/fhir/bucket/app/Main.java:999]: ok, i pushed a commit that does this...let me know what you think

**Review by punktilious — APPROVED** [review]
LGTM

**Pull request merged**